File: src/0242.py
from collections import defaultdict

class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t):
            return False
        ana = defaultdict(int)
        for n in s:
            ana[n] += 1
        for m in t:
            ana[m] -= 1
        return not any(ana.values())

File: src/0242.cpp
#include <iostream>
#include <algorithm>
#include <unordered_map>

using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) {
            return false;
        }
        unordered_map<char, int> ana;
        for (char c : s) {
            ana[c]++;
        }
        for (char c : t) {
            ana[c]--;
        }
        for (auto kv : ana) {
            if (kv.second != 0) {
                return false;
            }
        }
        return true;
    }
};
